ntp-proto/src/keyset.rs

```diff
@@ -170,24 +164,27 @@ impl KeySet {
     }
 
     pub(crate) fn encode_cookie(&self, cookie: &DecodedServerCookie) -> Vec<u8> {
-        let mut plaintext = Self::plaintext(cookie);
-        let plaintext_len = plaintext.as_slice().len();
-
-        let (siv_tag, nonce) = self.keys[self.primary as usize]
-            .encrypt_in_place_detached(plaintext.as_mut(), &[])
+        let mut output = Self::plaintext(cookie);
+        let plaintext_length = output.as_slice().len();
+
+        // Add space for header (4 + 2 bytes), tag (16 bytes) and nonce (16 bytes) plus some margin (16 bytes)
+        output.resize(output.len() + 2 + 4 + 16 + 16 + 16, 0);
```

Comment:
what is this margin? we should be able to know exactly how much space we need. Also what is the tag? we don't add the tag here (it's just header, nonce, and ciphertext)

Reply:
Fixed and clarified calculation.

As for tag, see above